ksql-api/src/test/java/io/confluent/ksql/api/ApiTest.java

```diff
@@ -319,70 +371,96 @@ public void shouldCloseQuery() throws Exception {
         .sendJsonObject(DEFAULT_PUSH_QUERY_REQUEST_BODY, responseFuture);
 
     // Wait for all rows in the response to arrive
-    assertTrue(waitUntil(() -> {
-      Buffer buff = writeStream.getBody();
+    assertThatEventually(() -> {
       try {
+        Buffer buff = writeStream.getBody();
         QueryResponse queryResponse = new QueryResponse(buff.toString());
-        return queryResponse.rows.size() == DEFAULT_ROWS.size();
+        return queryResponse.rows.size();
       } catch (Throwable t) {
-        return false;
+        return Integer.MAX_VALUE;
       }
-    }));
+    }, is(DEFAULT_ROWS.size()));
 
     // The response shouldn't have ended yet
-    assertFalse(writeStream.isEnded());
+    assertThat(writeStream.isEnded(), is(false));
 
     // Assert the query is still live on the server
     QueryResponse queryResponse = new QueryResponse(writeStream.getBody().toString());
     String queryId = queryResponse.responseObject.getString("queryId");
-    assertTrue(server.getQueryIDs().contains(new PushQueryId(queryId)));
-    assertEquals(1, server.getQueryIDs().size());
-    assertEquals(1, testEndpoints.getQueryPublishers().size());
+    assertThat(server.getQueryIDs().contains(new PushQueryId(queryId)), is(true));
+    assertThat(server.getQueryIDs(), hasSize(1));
+    assertThat(testEndpoints.getQueryPublishers(), hasSize(1));
 
     // Now send another request to close the query
     JsonObject closeQueryRequestBody = new JsonObject().put("queryId", queryId);
     HttpResponse<Buffer> closeQueryResponse = sendRequest(client, "/close-query",
         closeQueryRequestBody.toBuffer());
-    assertEquals(200, closeQueryResponse.statusCode());
+    assertThat(closeQueryResponse.statusCode(), is(200));
 
     // Assert the query no longer exists on the server
-    assertFalse(server.getQueryIDs().contains(new PushQueryId(queryId)));
-    assertEquals(0, server.getQueryIDs().size());
-    assertEquals(1, testEndpoints.getQueryPublishers().size());
-    assertFalse(testEndpoints.getQueryPublishers().iterator().next().hasSubscriber());
+    assertThat(server.getQueryIDs().contains(new PushQueryId(queryId)), is(false));
```

Comment:
FYI, can be:

```suggestion
    assertThat(server.getQueryIDs(), not(hasItem(new PushQueryId(queryId))));
```

Uses `not()` to invert the matcher.